mlflow/pyspark/optuna/study.py

```diff
@@ -0,0 +1,188 @@
+import datetime
+import logging
+import sys
+from collections.abc import Callable, Iterable
+from typing import Optional
+
+try:
+    import optuna
+    from optuna import pruners, samplers, storages
+    from optuna.study import Study
+    from optuna.study._optimize import _optimize_sequential
+    from optuna.trial import FrozenTrial
+except ImportError:
+    sys.exit()
+
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, first
+
+import mlflow
+from mlflow.exceptions import ExecutionException
+from mlflow.optuna.storage import MlflowStorage
+
+logger = logging.getLogger("optuna-spark")
+
+
+def is_spark_connect_mode():
+    """Check if the current Spark session is running in client mode."""
+    try:
+        from pyspark.sql.utils import is_remote
+    except ImportError:
+        return False
+    return is_remote()
+
+
+class MlflowSparkStudy(Study):
+    """A wrapper of :class:`~optuna.study.Study` to incorporate Optuna with spark
+    via MLflow experiment.
+
+    .. code-block:: python
+        :caption: Example
+
+        from mlflow.optuna.storage import MlflowStorage
+        from mlflow.pyspark.optuna.study import MlflowSparkStudy
+
+
+        def objective(trial):
+            x = trial.suggest_float("x", -10, 10)
+            return (x - 2) ** 2
+
+
+        experiment_id = "507151065975140"
+        study_name = "spark_mlflow_storage"
+
+        storage = MLFlowStorage(experiment_id=experiment_id)
+        mlflow_study = MlflowSparkStudy(
+            study_name, storage, mlflow_tracking_uri=mlflow.get_tracking_uri()
+        )
+        mlflow_study.optimize(objective, n_trials=4)
+    """
+
+    def __init__(
+        self,
+        study_name: str,
+        storage: MlflowStorage,
+        sampler: Optional["samplers.BaseSampler"] = None,
+        pruner: Optional[pruners.BasePruner] = None,
+        mlflow_tracking_uri: Optional[str] = "databricks",
+    ):
+        self.study_name = study_name
+        self._storage = storages.get_storage(storage)
+        self.sampler = sampler or samplers.TPESampler()
+        self.pruner = pruner or pruners.MedianPruner()
+
+        self.spark = SparkSession.active()
+
+        # check whether the SparkConnect mode
+        self._is_spark_connect_mode = is_spark_connect_mode()
+        self._mlflow_tracking_env = mlflow_tracking_uri
+
+        mlflow.set_tracking_uri(self._mlflow_tracking_env)
```

Comment:
```suggestion
        self.client = MlflowClient(self._mlflow_tracking_env)
```

let's use the client.